lib/SourceMapDevToolPlugin.js

```diff
@@ -38,128 +38,126 @@ SourceMapDevToolPlugin.prototype.apply = function(compiler) {
 		compilation.plugin("after-optimize-chunk-assets", function(chunks) {
 			var allModules = [];
 			var allModuleFilenames = [];
-			var tasks = [];
-			chunks.forEach(function(chunk) {
-				chunk.files.filter(ModuleFilenameHelpers.matchObject.bind(undefined, options)).map(function(file) {
-					var asset = this.assets[file];
-					if(asset.__SourceMapDevToolData) {
-						var data = asset.__SourceMapDevToolData;
-						for(var cachedFile in data) {
-							this.assets[cachedFile] = data[cachedFile];
-							if(cachedFile !== file)
-								chunk.files.push(cachedFile);
+			function getSourceAndMap(asset) {
```

Comment:
I think this function technically doesn't need to be in this scope. It can be moved up to like line 35 or so? It doesn't have any closure dependencies exception `options` afaict.